packages/next/build/index.ts

```diff
@@ -262,13 +263,16 @@ export default async function build(dir: string, conf = null): Promise<void> {
     isLikeServerless ? SERVERLESS_DIRECTORY : SERVER_DIRECTORY,
     PAGES_MANIFEST
   )
+  const buildManifestPath = path.join(distDir, BUILD_MANIFEST)
 
   const sprPages = new Set<string>()
   const staticPages = new Set<string>()
   const invalidPages = new Set<string>()
   const hybridAmpPages = new Set<string>()
   const pageInfos = new Map<string, PageInfo>()
   const pagesManifest = JSON.parse(await fsReadFile(manifestPath, 'utf8'))
+  const buildManifest = JSON.parse(await fsReadFile(buildManifestPath, 'utf8'))
+
```

Comment:
Should we pre-stat the file sizes here and pass them down since multiple pages could have common dependencies and then we're doing extra `fsStat` calls in `getPageSizeInKb` right?

Reply:
I memoized the fs.stat results below. I thought that would be useful in case another function wants to use fsStat too, and we won't need to add the additional logic of choosing files for modern here. Let me know if this works

Reply:
Sounds good 👍